app/javascript/components/project-submissions/components/flag-form.jsx

```diff
@@ -30,6 +30,15 @@ const FlagForm = ({ onSubmit, submission, userId }) => {
     );
   }
 
+  if (submission.id === undefined || submission.id === null) {
```

Comment:
Hi @rankoliang, what was the reason for adding this check?  

I don't think a user will ever be able to access this component with a submission with no id and the backend should already handle this this error in the event that user submits a flag for a submission shortly after its been deleted if I recall correctly :)

Reply:
Still a bit new to this, so I did the check just in case, if I am to be honest. I figured in the case it ever does happen, it would be better to catch it before the user spends their time to submit a report that is guaranteed to fail. I can revert that commit though if you still find it unnecessary though!

Reply:
ahh gotcha, yeah I think we can do without this check since the backend should cover it.